node/src/services/blockchain/transaction.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    str::FromStr,
+    time::{Duration, Instant},
+};
+
+use anyhow::anyhow;
+use log::*;
+use sp_core::H256;
+use storage_hub_infra::actor::ActorHandle;
+use tokio::sync::mpsc::Receiver;
+
+use crate::services::blockchain::{commands::BlockchainServiceInterface, types::ExtrinsicResult};
+
+use super::{types::ExtrinsicHash, BlockchainService};
+
+const LOG_TARGET: &str = "blockchain-transaction";
+
+#[derive(Debug)]
+pub struct SubmittedTransaction {
+    watcher: Receiver<String>,
+    hash: ExtrinsicHash,
+    timeout: Option<Duration>,
+}
```

Comment:
It'd be good to have some doc comments, specially explaining that this `watcher` is the receiver side of a channel coming from the `RpcHandlers` response. 
The reason I'm asking is that there are a lot of indirections due to multiple channels of communication, and it's docs here would reduce the overhead of understanding what is coming from where.